frontend/src/components/modules/Tournaments/MatchButton.tsx

```diff
@@ -0,0 +1,244 @@
+  FormControl,
+  InputLabel,
+  type ButtonProps
+} from "@mui/material";
+import type { TournamentPlayer } from "./OngoingTournament/RoundRobin/RoundRobinTournamentView";
+import type { Match, Tournament } from "types/models";
+import PlayerName from "./PlayerNames";
+import api from "api/axios";
+import type { ChangeCourtTimeRequest } from "types/requests";
+
+interface MatchButtonProps {
+  match: Match;
+  players: TournamentPlayer[];
+  navigate: (path: string) => void;
+  t: (key: string) => string;
+  haveSameNames: boolean;
+  props: ButtonProps;
+  isUserTheCreator: boolean;
+  tournamentData: Tournament;
+}
+
+const MatchButton: React.FC<MatchButtonProps> = ({
+  match,
+  players,
+  navigate,
+  t,
+  haveSameNames,
+  props,
+  isUserTheCreator,
+  tournamentData
+}) => {
+  // State to track form input and modal open state
+  const [editMode, setEditMode] = useState(false);
+  const [newCourtNumber, setNewCourtNumber] = useState<number>(
+    match.courtNumber
+  );
+  const [newTime, setNewTime] = useState(
+    match.scheduledTime === "XX:XX" ? "00:00" : match.scheduledTime
+  );
+
+  // Function to handle modal open/close
+  const handleOpen = (): void => {
+    setEditMode(true);
+  };
+
+  const handleClose = (): void => {
+    setEditMode(false);
+  };
+
+  // Function to handle change court time form submit
+  const handleSubmit = async (): Promise<void> => {
+    const updates: Partial<ChangeCourtTimeRequest> = {};
+
+    // Check if the court number has changed
+    if (Number(newCourtNumber) !== match.courtNumber) {
+      updates.courtNumber = Number(newCourtNumber);
+    }
+
+    // Check if the scheduled time has changed
+    if (newTime !== match.scheduledTime) {
+      updates.scheduledTime = newTime;
+    }
+
+    // If there are updates, send the request
+    if (Object.keys(updates).length > 0) {
+      try {
+        await api.match.changeCourtAndTime(match.id, updates);
+        console.log("Court and time updated successfully");
+      } catch (error) {
+        console.error("Error updating court and time:", error);
+      }
+    } else {
+      console.log("No changes detected, no update needed");
+    }
+
+    setEditMode(false); // Close modal after submit
+  };
+
+  // Find the players in the players array using their IDs
+  const player1 = players.find(
+    (player) => player.id === match.players[0].id
+  ) as TournamentPlayer;
+  const player2 = players.find(
+    (player) => player.id === match.players[1]?.id
+  ) as TournamentPlayer;
+
+  // Get the names of the players
+  const player1Name = (
+    <PlayerName
+      firstName={player1.firstName}
+      lastName={player1.lastName}
+      sameNames={haveSameNames}
+    />
+  );
+  const player2Name =
+    player2 !== undefined ? (
+      <PlayerName
+        firstName={player2.firstName}
+        lastName={player2.lastName}
+        sameNames={haveSameNames}
+      />
+    ) : (
+      <PlayerName firstName="BYE" lastName="" sameNames={false} />
```

Comment:
This not needed for player 1?

Reply:
Could be added to both, but should never happen because when registering the names have to be set. This part was taken from the previous groups code and just moved to its own component.

Reply:
I dont think it matters if it works then but which ever seems more wise :D